File: solutions/2026-05-11/cpp/2026-05-11-separate-the-digits-in-an-array.cpp
```cpp
// Problem: Separate the Digits in an Array
// Link: https://leetcode.com/problems/separate-the-digits-in-an-array/
//
// Approach:
// We will iterate through each number in the input array `nums`.
// For each number, we need to extract its digits. A common way to do this
// is by repeatedly taking the number modulo 10 to get the last digit,
// and then dividing the number by 10 to remove the last digit, until the
// number becomes 0. Since we need the digits in the same order they appear
// in the original number, we can store these digits in a temporary list
// and then reverse it before adding to the final answer. Alternatively,
// we can convert the number to a string and then iterate through the
// characters of the string, converting each character back to an integer.
// This string conversion approach is generally simpler and more direct for
// this problem.
//
// Time Complexity:
// Let N be the number of elements in `nums` and K be the maximum number
// of digits in any element of `nums`.
// For each number in `nums`, we convert it to a string, which takes O(K) time.
// Then, we iterate through the string, which also takes O(K) time.
// Therefore, for each number, the operation takes O(K) time.
// The total time complexity is O(N * K). Given the constraints (nums[i] <= 10^5),
// K is at most 6. So, the time complexity is effectively O(N).
//
// Space Complexity:
// We use an `answer` vector to store the separated digits. In the worst case,
// if each digit is a separate number, the size of `answer` can be up to N * K.
// The space complexity is O(N * K). Similarly, due to the constraints on `nums[i]`,
// this is effectively O(N).
// If we consider the space used by the string conversion, it also takes O(K)
// for each number, contributing to the overall space complexity.

#include <vector>
#include <string>
#include <algorithm>

class Solution {
public:
    std::vector<int> separateDigits(std::vector<int>& nums) {
        // Initialize an empty vector to store the separated digits.
        std::vector<int> answer;

        // Iterate through each number in the input array `nums`.
        for (int num : nums) {
            // Convert the current number to its string representation.
            std::string s = std::to_string(num);

            // Iterate through each character (digit) of the string.
            for (char c : s) {
                // Convert the character back to an integer and add it to the `answer` vector.
                // Subtracting '0' from a character digit converts it to its integer equivalent.
                // For example, '1' - '0' = 1.
                answer.push_back(c - '0');
            }
        }

        // Return the vector containing all separated digits.
        return answer;
    }
};
```

File: solutions/2026-05-11/python/2026-05-11-separate-the-digits-in-an-array.py
```python
# Problem: Separate the Digits in an Array
# Link: https://leetcode.com/problems/separate-the-digits-in-an-array/
# Approach:
# Iterate through each number in the input array `nums`.
# For each number, convert it to a string to easily access its digits.
# Then, iterate through the characters of the string.
# Convert each character back to an integer and append it to the `answer` list.
# This ensures that the digits are added in the same order they appear in the original number.
# Time Complexity: O(N * log10(max(nums))), where N is the number of elements in `nums`.
# The `log10(max(nums))` factor comes from converting each number to a string,
# which depends on the number of digits in the largest number.
# Space Complexity: O(D), where D is the total number of digits across all numbers in `nums`.
# This is the space required to store the `answer` list.

class Solution:
    def separateDigits(self, nums: list[int]) -> list[int]:
        # Initialize an empty list to store the separated digits.
        answer = []

        # Iterate through each number in the input list `nums`.
        for num in nums:
            # Convert the current number to a string to easily access its digits.
            num_str = str(num)
            # Iterate through each character (digit) in the string representation of the number.
            for digit_char in num_str:
                # Convert the digit character back to an integer and append it to the `answer` list.
                answer.append(int(digit_char))

        # Return the list containing all separated digits.
        return answer

```